src/main/java/com/matera/account/account/AccountService.java

```diff
@@ -40,11 +42,25 @@ public Account updateAccountByAccountId(UUID clientId, UUID accountId, AccountDT
         account.setAccountTypeId(accountDTO.getAccountTypeId());
         account.setAgency(accountDTO.getAgency());
         account.setBalance(accountDTO.getBalance());
-        account.setClientId(accountDTO.getClientId());
         accountRepository.save(account);
         return account;
     }
 
+    public Account patchAccountByAccountId(UUID clientId, UUID accountId, AccountDTO accountDTO) {
+        Optional<Account> account = accountRepository.findByAccountIdAndClientId(clientId, accountId);
+        account.orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Couldn't Found Account"));
+        if (StringUtils.isNotBlank(accountDTO.getAccountNumber()))
+            account.get().setAccountNumber(accountDTO.getAccountNumber());
+        if (accountDTO.getAccountTypeId() != null)
+            account.get().setAccountTypeId(accountDTO.getAccountTypeId());
+        if (StringUtils.isNotBlank(accountDTO.getAgency()))
+            account.get().setAgency(accountDTO.getAgency());
+        if (accountDTO.getBalance() != null)
+            account.get().setBalance(accountDTO.getBalance());
+        accountRepository.save(account.get());
```

Comment:
you can change these two lines to `return accountRepository.save(account.get());` the `save` method returns the entity :)